modules/feature-management/src/main/java/org/mule/runtime/feature/internal/togglz/provider/DefaultMuleTogglzFeatureProvider.java

```diff
@@ -67,11 +65,11 @@ public void addFeatureEnum(Class<? extends Feature> featureEnum) {
   }
 
   private void addFeatures(Collection<? extends Feature> newFeatures) {
```

Comment:
can this method be executed concurrently? If so, the initialization of `metaDataCache` is broken

Reply:
Not currently, as this is used for adding the default enum features in togglz at the beginning. But I'll synchronize it because of future uses. The case for addTogglzFeatureMetadata could fail in parallel deployment. I've synchronized it.